src/domains/wallet-app/api/npg-notifications/v1/_base_policy.xml.tpl

```diff
@@ -1,15 +1,54 @@
 <policies>
     <inbound>
       <base />
+      <set-variable name="orderIdPathParam" value="@(context.Request.MatchedParameters["orderId"])" />
+      <set-variable name="walletId" value="@(context.Request.MatchedParameters["walletId"])" />
+      <set-variable name="npgNotificationRequestBody" value="@((JObject)context.Request.Body.As<JObject>(true))" />
+      <set-variable name="orderIdBodyParam" value="@((string)((JObject)requestBody["operation"])["orderId"])" />
+      <choose>  
+        <when condition="@((string)context.Variables["orderIdPathParam"] != (string)context.Variables["orderIdBodyParam"])">
+          <return-response>
+            <set-status code="400" reason="orderId mismatch" />
+            <set-header name="Content-Type" exists-action="override">
+                <value>application/json</value>
+            </set-header>
+            <set-body>{
+                    "title": "Unable to handle notification",
+                    "status": 400,
+                    "detail": "orderId mismatch",
+                }</set-body>
+          </return-response>
+        </when>
+      </choose>  
+      <set-body>
+          @{
+            JObject requestBody = (JObject)context.Variables["npgNotificationRequestBody"];
+            JObject operation = (JObject)requestBody["operation"];
+            string operationResult = (string)operation["operationResult"];
+            string operationId = (string)operation["operationId"];
```

Comment:
We can eliminate this

Reply:
for now we will use it as the npg notification ID